pipelines/folioRancher/folioQualityGates/Jenkinsfile

```diff
@@ -27,20 +29,36 @@ pipeline {
             }
         }
 
-        stage('Check Changes') {
-            steps {
-                script {
-                    if (params.SKIP_CHANGES == true) {
-                        echo "Skipping changes For the SNAPSHOT BRANCH and moving to Parrarel Run ."
-                    } else {
-                        checkout scm
-                        echo "Checking Changes for the SNAPSHOT BRANCH..and moving to Parrarel Run ."
-                    }
+stage('Check Changes') {
+    steps {
+        script {
+           if (params.SKIP_CHANGES) {
+                echo "Skipping changes for the SNAPSHOT BRANCH and moving to Parallel Run."
+            } else {
+                withCredentials([[$class           : 'AmazonWebServicesCredentialsBinding',
+                                  credentialsId    : Constants.AWS_CREDENTIALS_ID,
+                                  accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+                                  secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+
+                def HashDiffDetected = folioHashCommitCheck.commitHashChangeDetected(params.BRANCH)  // true or false
+                echo "HashCommitChanges: ${HashDiffDetected}"
+                env.HashDiffDetected = HashDiffDetected.toString()
+                if (HashDiffDetected) {
+                   // Set the build status to UNSTABLE (Yellow) To Indicate that Changes Are Detected, For the Visualisation Purposes
+                   echo "Changes detected, setting build status to UNSTABLE"
+                   currentBuild.result = 'UNSTABLE'
```

Comment:
So i think that clear code here should looks like
env.DIFF_DETECTED=folioHashCommitCheck.commitHashChangeDetected(params.BRANCH).toString()

Reply:
thanks , good one ,i will use like this